Source/Particles/MultiParticleContainer.cpp

```diff
@@ -948,4 +974,137 @@ MultiParticleContainer::BreitWheelerGenerateTable ()
+            {
+                auto& src_tile = pc_source ->ParticlesAt(lev, mfi);
+                auto& dst_ele_tile = pc_product_ele->ParticlesAt(lev, mfi);
+                auto& dst_pos_tile = pc_product_pos->ParticlesAt(lev, mfi);
+
+                const auto np_dst_ele = dst_ele_tile.numParticles();
+                const auto np_dst_pos = dst_pos_tile.numParticles();
+                const auto num_added = filterCopyTransformParticles<1>(
+                    dst_ele_tile, dst_pos_tile,
+                    src_tile, np_dst_ele, np_dst_pos,
+                    Filter, CopyEle, CopyPos, Transform);
+
+                setNewParticleIDs(dst_ele_tile, np_dst_ele, num_added);
+                setNewParticleIDs(dst_pos_tile, np_dst_pos, num_added);
+            }
+
+
+
+        }
+   }
+}
+
+void MultiParticleContainer::doQedQuantumSync()
+{
+    WARPX_PROFILE("MPC::doQedEvents::doQedQuantumSync");
+
+    // Loop over all species.
+    // Electrons or positrons undergoing Quantum photon emission process
+    // create photons in pc_product_phot
+
+    for (auto& pc_source : allcontainers){
+        if(!pc_source->has_quantum_sync()){ continue; }
+
+                // Get product species
+        auto& pc_product_phot =
+            allcontainers[pc_source->m_qed_quantum_sync_phot_product];
+
+        SmartCopyFactory copy_factory_phot(*pc_source, *pc_product_phot);
+        auto phys_pc_ptr =
+            static_cast<PhysicalParticleContainer*>(pc_source.get());
+
+        const auto Filter    = phys_pc_ptr->getPhotonEmissionFilterFunc();
+        const auto CopyPhot   = copy_factory_phot.getSmartCopy();
+
+        auto Transform = PhotonEmissionTransformFunc(
+            m_shr_p_qs_engine->build_optical_depth_functor(),
+            pc_source->particle_runtime_comps["optical_depth_QSR"],
+             m_shr_p_qs_engine->build_phot_em_functor());
+
+        pc_source ->defineAllParticleTiles();
+        pc_product_phot->defineAllParticleTiles();
+
+        for (int lev = 0; lev <= pc_source->finestLevel(); ++lev)
+        {
+            const auto info = getMFItInfo(*pc_source, *pc_product_phot);
+
+#ifdef _OPENMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+            for (MFIter mfi = pc_source->MakeMFIter(lev, info); mfi.isValid(); ++mfi)
+            {
+                auto& src_tile = pc_source->ParticlesAt(lev, mfi);
+                auto& dst_tile = pc_product_phot->ParticlesAt(lev, mfi);
+
+                const auto np_dst = dst_tile.numParticles();
+
+                const auto num_added =
+                    filterCopyTransformParticles<1>(dst_tile, src_tile, np_dst,
+                        Filter, CopyPhot, Transform);
+
+                setNewParticleIDs(dst_tile, np_dst, num_added);
+
+                constexpr auto energy_threshold = static_cast<amrex::ParticleReal>(
+                     2.0 * PhysConst::m_e * PhysConst::c * PhysConst::c);
```

Comment:
Should this be an input parameter?

Reply:
yes, we could make it an input parameter... I will do that

Reply:
I should have implemented the change. The energy threshold for photon creation is now an
optional input parameter. I have updated the documentation accordingly.